sunkit_image/granule.py

```diff
@@ -0,0 +1,331 @@
+    "kmeans_segment",
+    "cross_correlation",
+]
+
+
+def segment(data_map, skimage_method, res, mark_dim_centers=False):
+    """
+    Segment optical image of the solar photosphere into tri-value maps
+    with 0 = intergranule, 0.5 = faculae, 1 = granule.
+
+    Parameters
+    ----------
+    data_map : `SunPy map`
+        SunPy map containing data to segment
+    skimage_method : `str`
+        skimage thresholding method, with options 'otsu', 'li', 'isodata',
+        'mean', 'minimum', 'yen', and 'triangle'.
+        Note - depending on input data, one or more of these methods may be
+        signifcantly better or worse than the others. Typically, 'li', 'otsu',
+        'mean', and 'isodata' are good choices, 'yen' and 'triangle' over-
+        identify intergranule material, and 'minimum' overidentifies granules.
+    res : `float`
+        Spatial resolution (arcsec/pixel) of the data
+    mark_dim_centers : `bool`
+        Whether to mark dim granule centers as a seperate catagory for future
+        exploration.
+
+    Returns
+    -------
+    segmented_map : `SunPy map`
+        SunPy map containing segmentated image (with the original header)
+    """
+
+    if not isinstance(data_map, sunpy.map.mapbase.GenericMap):
+        raise TypeError("Input must be sunpy map.")
+
+    methods = ["otsu", "li", "isodata", "mean", "minimum", "yen", "triangle"]
+    if skimage_method not in methods:
+        raise TypeError("Method must be one of: " + str(methods))
+
+    data = data_map.data
+    header = data_map.meta
+
+    # apply median filter
+    median_filtered = sndi.median_filter(data, size=3)
+
+    # apply threshold
+    threshold = get_threshold(median_filtered, skimage_method)
+
+    # initial skimage segmentation
+    segmented_image = np.uint8(median_filtered > threshold)
+
+    # fix the extra IGM bits in the middle of granules
+    seg_im_fixed = trim_intergranules(segmented_image, mark=mark_dim_centers)
+
+    # mark faculae and get final granule and facule count
+    seg_im_markfac, fac_cnt, gran_cnt = mark_faculae(seg_im_fixed, data, res)
+    logging.info("Segmentation has identified " + str(gran_cnt) + " granules and " + str(fac_cnt) + "faculae")
+
+    # convert segmentated image back into SunPy map with original header
+    segmented_map = sunpy.map.Map(seg_im_markfac, header)
+
+    return segmented_map
+
+
+def get_threshold(data, method):
+    """
+    Get the threshold value using given skimage segmentation type.
+
+    Parameters
+    ----------
+    data : `numpy.ndarray`
+        Data to threshold.
+    method : `str`
+        Skimage thresholding method - options are 'otsu', 'li', 'isodata',
+        'mean', 'minimum', 'yen', 'triangle'.
+
+    Returns
+    -------
+    threshold `float`:
+        Threshold value.
+    """
+
+    if not isinstance(data, np.ndarray):
+        raise ValueError("Input data must be an array.")
+
+    methods = ["otsu", "li", "isodata", "mean", "minimum", "yen", "triangle"]

```

Comment:
I think since this list has been defined here and in the main function, we could pull it out to be a constant at the top of the file instead.